modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapperTestCase.java

```diff
@@ -47,6 +58,23 @@ public void recursiveParameterType() {
     }
   }
 
+  @Test
+  @Issue("W-12622240")
+  @Description("Ensure JDK internal fields are filtered out as we can't use reflection on them in Java 17 and above")
+  public void filterInternalJdkFields() {
+    TypeWrapper type = new TypeWrapper(LocalDateTime.class, new DefaultExtensionsTypeLoaderFactory()
+        .createTypeLoader(currentThread().getContextClassLoader()));
+    List<FieldElement> fields = type.getFields();
+    List<String> allPackages =
+        type.getFields().stream().map(f -> f.getField().get().getDeclaringClass().getPackage().getName()).collect(toList());
+    assertThat(fields, empty());
```

Comment:
if this assertion is true, what is the point on iterating this and assert on the result of the iteration?